test/InternalServerTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+use donatj\MockWebServer\InternalServer;
+use donatj\MockWebServer\MockWebServer;
+
+class InternalServerTest extends \PHPUnit_Framework_TestCase
+{
+
+    private $testTmpDir;
+    private $server;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->testTmpDir = __DIR__ . DIRECTORY_SEPARATOR . 'testTemp';
+
+        mkdir($this->testTmpDir);
+
+        $counterFileName = $this->testTmpDir . DIRECTORY_SEPARATOR . MockWebServer::REQUEST_COUNT_FILE;
+
+        file_put_contents($counterFileName, '0');
+    }
+
+    public function tearDown()
+    {
+        parent::tearDown();
+        $this->removeTempDirectory();
+    }
+
+    private function removeTempDirectory()
+    {
+        $it = new RecursiveDirectoryIterator($this->testTmpDir, RecursiveDirectoryIterator::SKIP_DOTS);
+        $files = new RecursiveIteratorIterator($it,
+            RecursiveIteratorIterator::CHILD_FIRST);
+
+        foreach ($files as $file) {
+            if ($file->isDir()) {
+                rmdir($file->getRealPath());
+            } else {
+                unlink($file->getRealPath());
+            }
+        }
+
+        rmdir($this->testTmpDir);
+    }
+
+    /**
+     * @param $inputCount
+     * @param $expectedCount
+     *
+     * @dataProvider countProvider
+     */
+    public function testShouldIncrementRequestCounter($inputCount, $expectedCount)
+    {
+        $counterFileName = $this->testTmpDir . DIRECTORY_SEPARATOR . MockWebServer::REQUEST_COUNT_FILE;
+
+        InternalServer::incrementRequestCounter($this->testTmpDir, $inputCount);
+
+        $this->assertEquals(file_get_contents($counterFileName), $expectedCount);
+
+    }
+
+    public function countProvider()
+    {
+        return [
+            'null count' => [
+                'inputCount' => null,
+                'expectedCount' => 1,
+            ],
+            'int count' => [
+                'inputCount' => 25,
+                'expectedCount' => 25,
+            ]
+        ];
+    }
+
+    public function testShouldLogRequestsOnInstanceCreate()
+    {
```

Comment:
A change I'd request throughout right off the bat is that you've indented with spaces and put braces on the next line. Different than the rest of the project.

I go for tabs and braces on the same line.  

If you're interested in correcting it, that would be great. I can correct it myself otherwise ;)

Reply:
Thanks for the review. Of course, I will change the code style.